File: frontend/src/api/smtpConfig.ts
import client from './client'

export interface SmtpConfig {
  host: string
  port: number
  username: string
  from_addr: string
  use_tls: boolean
}

export interface SmtpConfigUpdate extends Partial<SmtpConfig> {
  password?: string
}

export const getSmtpConfig = () => client.get<SmtpConfig>('/smtp-config')
export const updateSmtpConfig = (data: SmtpConfigUpdate) => client.patch<SmtpConfig>('/smtp-config', data)

File: backend/app/schemas/smtp_config.py
from pydantic import BaseModel


class SmtpConfigOut(BaseModel):
    host: str
    port: int
    username: str
    from_addr: str
    use_tls: bool


class SmtpConfigUpdate(BaseModel):
    host: str | None = None
    port: int | None = None
    username: str | None = None
    password: str | None = None
    from_addr: str | None = None
    use_tls: bool | None = None
